src/agents/user_agent.py

```diff
@@ -76,31 +83,33 @@ async def request_reservation_modification(self, reservation_id, time_start, tim
             self.pending_reservation.time_start = time_start
             self.pending_reservation.time_stop = time_stop
             modify_reservation_msg = ModifyReservation(
-                reservation_id=reservation_id, time_start=time_start, time_stop=time_stop, user_id=self.jid
+                reservation_id=reservation_id, time_start=time_start, time_stop=time_stop, user_id=str(self.jid)
             )
+            # TODO critical: message.to should be correct regional coordinator
```

Comment:
Is this critical message solved?

Reply:
Is in the other PR ;) Maybe let's first merge this branch - agent-improvements to this one - sample-test-case. And then this one to main?